compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -362,10 +362,21 @@ private BSymbol findSymbolAtCursorPosition(Document sourceDocument, LinePosition
                     typesFactory.getTypeDescriptor(symbolAtCursor.type, symbolAtCursor));
         }
 
+        if (isTypeSymbol(symbolAtCursor) &&
```

Comment:
Yes I can move the logic inside. This is needed because for local anonymous error types the package symbol was previously coming from annotation packages `error type` defined in symbol table. Therefore we were relying on package id to identify locally defined error types. I changed the logic to consider `ANONYMOUS` flag.

Check the changes to API 
```public BType transform(BLangErrorType errorTypeNode, AnalyzerData data)```
in SymbolResolver.